### DISCUSSION TITLE
open rewrite upgrade scripts for Configurations and property files

### DISCUSSION THREAD

**Discussion by jjansenvr** [frankframework/frankframework#7409] [Ideas]
We are thinking of starting a trainee ship on creating open rewrite recipes for upgrades

Upgrading from 7.0 upward
https://docs.openrewrite.org/recipes/xml
What functional requirements do we have for this?

- replacing deprecated pipes, attributes and defaults
- getting rid of workarounds (like having transaction attributes on pipes)
- all console warnings

**Comment by nielsm5** [comment]
Upgrading the config from old to new style xml?

**Reply by jjansenvr** [reply]
Would be a good addition
